### PR TITLE
Chore: Share templates between client and server

### LINKED ISSUE #244: Share templates between server and client
We have to find a way of sharing the `.hbs` of the scan result item and just have one copy.
Now if we want to update something in there we have to do it in 2 different places, the `.hbs` and `scanner-submit.js`

### PR DESCRIPTION
**Pull request by qzhou1607**
## Pull request checklist

Make sure you:

- [X] Signed the [Contributor License Agreement](https://cla.js.foundation/sonarwhal/sonar)
- [X] Followed the [commit message guidelines](https://sonarwhal.com/docs/developer-guide/contributing/pull-requests.html#commitmessages)

## Short description of the change(s)
Fix #244

1. Register a hexo renderer that retrieves partials from the `partial` folder and inserts the templates in the js scripts.
2. Refactor `.hbs` files so that the client and server sides are able to share partials.
3. Refactor helpers so that the client and server could share helpers as well.
4. Highlight code block when results are rendered from the sever side.
5. Remove the templates/helpers in the client side.

### CODE REVIEW

**Comment by molant** [comment]
@qzhou1607 there are conflicts in this PR. Can you please solve them?

**Comment by qzhou1607** [comment]
@molant Done.

**Comment by molant** [comment]
@qzhou1607 conflicts again, I promise I won't merge anything else until this one gets into 🙏

**Comment by qzhou1607** [comment]
@molant In the middle of confirming everything is okay, please don't merge yet 😄

**Comment by qzhou1607** [comment]
@molant This is ready. I also updated the links to documentation in the scan result page, the changes there got lost somehow after the normalization url PR was merged.

**Comment by molant** [comment]
@qzhou1607 looks like there are still conflicts in here. Maybe something that @alrra merged over the weekend? I promise it wasn't me!

**Comment by alrra** [comment]
> Maybe something that @alrra merged over the weekend? 

[People needed their docs!](https://github.com/sonarwhal/sonarwhal.com/commit/e73f0e60dbdced70ac727ecb236e7cd752a6b5fd)

**Comment by qzhou1607** [comment]
@molant  Done! 😂 
@alrra Thank you for taking care of that bug!

File: src/server/routes/scanner.js [hunk 1]
```diff
@@ -149,6 +152,20 @@ const processRuleResults = (ruleResults, scanUrl) => {
...
 };
 
+const renderHelpers = (req, res, next) => {
+    const requiredHelpers = req.params.name.split(',');
+
+    if (requiredHelpers.length === 0) {
+        return next();
+    }
+
+    res.type('application/javascript');
```
Comment (molant) [inline src/server/routes/scanner.js]: `text/javascript`: https://github.com/whatwg/html/pull/3096

**Comment by qzhou1607** [comment]
@molant Yes. Helpers such as `sortPagesByCategory` is not used in the scanner client side and only used when rendering doc pages. What I like about the endpoint is that we have better control over what helpers are loaded in a page, otherwise we will have to load all the helper functions even though we don't need all of them in the client side. What do you think?

File: src/server/routes/scanner.js (hunk 1 above)
Comment (qzhou1607) [reply src/server/routes/scanner.js]: @molant  Fixed.

**Comment by molant** [comment]
> otherwise we will have to load all the helper functions even though we don't need all of them in the client side

That's why I ask if we were loading all or not 😊

**Pull request merged**